### DISCUSSION TITLE
MQTT - writable sensors (HA integration?) and how to use them?

### DISCUSSION THREAD

**Discussion by stranskj** [john30/ebusd#1060] [Q&A]
Hi,
how can I find out, which sensors are writable?

I am not sure how to use them
I do have the `^w` flag in ebusd integration configuation file. I am watching the messages&topic in MQTTexplorer. All the sensors exposed into the `homeassistant` topic, seems to have just a configuration. For example, under `homeassistant/sensor/ebusd_bai_SetMode_disablehc/config`, I see this:
```
{
  "unique_id":"ebusd_bai_SetMode_disablehc",
  "name":"SetMode disablehc",
  "device":{
   "identifiers":"ebusd_bai",
   "manufacturer":"ebusd.eu",
   "name":"ebusd bai",
   "via_device":"ebusd",
   "sw_version":"23.2",
   "suggested_area":"Heating"
  },
  "value_template":"{{value_json[\"disablehc\"].value}}",
  "state_topic":"ebusd/bai/SetMode"
 }
```
I think, this is one of those I have not seen without `^w`, so I should be able to use it to change boiler parameter (e.g. turn heating on, I suppose). I can see a status read. But how can I set it/write to it? Should I publish to `ebusd/bai/SetMode` or `homeassistant/sensor/ebusd_bai_SetMode_disablehc/config` or anywhere else?

**Comment by Koky05** [comment]
In your bai configuration or on git you can check values (if it start with "w" it is writable):

> r2,,RoomTemp,RaumtemperaturIst,,,,
> r2;w,,RoomTempCorrection,RoomTempCorrection,,,

From HA MQTT devices you can see if it is writtable:
![image](https://github.com/john30/ebusd/assets/89223022/fb0ed71d-bff9-4c75-81a7-a404ceb7fdc0)

All of the homeassistant/sensor are not writtable (MQTT sensor doen't support command_topic). So you neet to search for:
homeassistant/number
homeassistant/select
homeassistant/text

They all contain in config, line with "command_topic":
`{
  "unique_id": "ebusd_f47_3_ccTimer_Friday_0",
  "name": "ccTimer.Friday from",
  "device": {
    "identifiers": "ebusd_f47_3",
    "manufacturer": "ebusd.eu",
    "name": "f47.3",
    "via_device": "ebusd",
    "sw_version": "23.2",
    "suggested_area": "Heating"
  },
  "value_template": "{{value_json[\"0\"].value}}",
  "state_topic": "ebusd/f47.3/ccTimer.Friday",
  "command_topic": "ebusd/f47.3/ccTimer.Friday/set",
  "pattern": "^[012][0-9]:[0-5][0-9]$"
}`

**Reply by stranskj** [reply]
I see... From what I saw elsewhere (zigbee2mqtt), I was expecting the `command_topic`, which I have not spotted in configs exposed by ebusd. So I was actually on a good track...

In my MQTT, I see only only sensors, no controls... Not even the data, which you show here...

Is there anywhere a list of the bai configuration lines, which is actually used? Finding them through the Git repo is very tedious...

**Comment by Koky05** [comment]
If you connect to ebus deamon (in my case ebusd docker container) and run command `ebusctl find -V -a` then you get all data you read with ebus adapter. Line with:
`bai clearerrorhistory = no data stored [ZZ=08, active write]` are writtable.

I use local copy of configuration files from:
[ebus-configuration](https://github.com/john30/ebusd-configuration/tree/master/ebusd-2.1.x/en/vaillant)

After call command `ebusctl i` I see that I use:

> loaded "vaillant/bai.0010015600.inc" ([HW=9602]), "vaillant/08.bai.csv"
> loaded "vaillant/15.f47.csv"
> loaded "vaillant/50.v61.mc.csv"

**Reply by stranskj** [reply]
Great, finally getting somewhere... Unfortunatelly, I see all message `active read`. Do I need to enable write somewhere, or I had a bad luck?

**Reply by stranskj** [reply]
I dont see the configuration csv anywhere in the container. Does it mean, that it always search them online?

**Reply by Koky05** [reply]
If you did not specify configpath then yes it use online configuration:

> environment:
>       - EBUSD_CONFIGPATH=/etc/ebusd/ebusd-configuration/ebusd-2.1.x/en


[config options](https://github.com/john30/ebusd/wiki/2.-Run#message-configuration-options)

**Reply by stranskj** [reply]
ok, thanks, I will look into that. It might be, that he online ones (I already found where it is), have messages read-only?

**Reply by stranskj** [reply]
So I loaded local config files, with no much more succes. I can see one `write active` (it is actually the date, but I cannot not find it under the `homeassistant` topic), and few `write passive`. And in those, I still dont see the `command_topic`

**Reply by Koky05** [reply]
If you want to see it you need also change "mqtt-hassio.cfg":
`filter-name = .....|date|timer|...`
`filter-direction = r|u|^w`

`# HA integration: currently unused mapping lists for non-numeric/non-binary entities.`
`#type_switch-string =`
`#    sensor,, =`
`type_switch-date =`
`    sensor,,measurement =`
`#type_switch-datetime =`
`#    sensor, =`



I don't remember all changes I made to see it.

**Reply by stranskj** [reply]
Well, no I can set the time... Probably. Nothing else....
How do the `write passive` work? Can I set values to those?
